Run ssl_test_old in fips

mattcaswell opened the pull request:
In a similar way to #11511 and #11508, we run ssl_test_old twice: once with a non-default library context with the default provider loaded into it, and once with a non-default library context with the FIPS provider loaded into it. In both cases we load the "null" provider into the default context to make sure we don't accidentally pick up algorithms from there.

These tests will fail since they require all the key gen PRs to be merged first as well as #11494 and #11507 (and I have not included them here). However, aside from the dependencies this should be fairly complete and can be reviewed.

levitte commented:
You might want to edit the subject, as #11328, #11303, #11332, and #11494 are now merged

levitte commented on test/recipes/80-test_ssl_old.t:
```diff
@@ -310,12 +329,14 @@ sub testss {
-		   "-c_key", $key, "-c_cert", $cert);
+		   "-c_key", $key, "-c_cert", $cert,
+           "-provider", $provider,
+           "-config", $configfile);
```
The new lines are unaligned.  The previous lines are indented with tabs, so it's possible it looks right to you if you have 4 space tab translation, but...  and github doesn't exactly help...

mattcaswell replied:
Fixed.

levitte commented on test/recipes/80-test_ssl_old.t:
```diff
@@ -13,12 +13,20 @@ use warnings;
+
+use lib srctop_dir('Configurations');
+use lib bldtop_dir('.');
+use platform;
```
You're not actually using `platform` anywhere that I can see...

levitte replied:
Oh, never mind that!  Saw it

mattcaswell commented:
> You might want to edit the subject, as #11328, #11303, #11332, and #11494 are now merged

Done

levitte commented on test/recipes/80-test_ssl_old.t:
```diff
@@ -78,10 +86,20 @@ my $client_sess="client.ss";
+            '-provider_name', 'fips', '-mac_name', 'HMAC',
+            '-macopt', 'digest:SHA256', '-macopt', 'hexkey:00',
+            '-section_name', 'fips_sect'])),
+   "fipsinstall");
```
Considering how this is copied all over the place, we might want to consider making this a separate script.

mattcaswell replied:
Yes...and a separate "make test" dependency so that we don't have to do this everywhere? Not *this* PR though I think.

levitte replied:
Oh!  That too

levitte replied:
Yeah, I'm getting to a point where I'll do this, in a separate PR

mattcaswell commented:
Rebased now that #11371 has been merged. I have had to include the same libssl fixup commit that I've added to #11508 here, in order to get the tests to pass.

*Please don't review the libssl changes in this PR. Please provide any review comments on that aspect in #11508 instead.*

I've taken this out of WIP, although it cannot now be pushed until #11508 goes in.

slontis commented:
Fixed the fips disabled config.

mattcaswell commented:
Ping - this needs review (and for the record I'm fine with @slontis's update to this PR).

mattcaswell commented:
Rebased now that #11508 has gone in. No other changes were made.

levitte commented on test/recipes/80-test_ssl_old.t:
```diff
@@ -13,13 +13,22 @@ use warnings;
+
+use lib srctop_dir('Configurations');
+use lib bldtop_dir('.');
+use platform;
```
You can skip this now...

levitte commented on test/recipes/80-test_ssl_old.t:
```diff
@@ -78,10 +87,21 @@ my $client_sess="client.ss";
+                '-macopt', 'digest:SHA256', '-macopt', 'hexkey:00',
+                '-section_name', 'fips_sect'])),
+       "fipsinstall");
+}
```
... and simplify this:

``` perl
unless ($no_fips) {
    ok(run(perltest(['fipsinstall.pl', bldtop_dir()])),
       'fipsinstall');
}
```

mattcaswell commented:
@levitte - I would prefer to push this as is, and modify the fipsinstall stuff with a follow on PR since this PR is now otherwise ready-to-merge.

openssl-machine commented:
24 hours has passed since 'approval: done' was set, but as this PR has been updated in that time the label 'approval: ready to merge' is not being automatically set.  Please review the updates and set the label manually.

slontis commented:
Considering there seemed to be some problem in travis when the fipsinstall stuff was done - that is probably a good idea.

levitte commented:
> with a follow on PR

Okie

mattcaswell commented:
Pushed. Thanks!

mattcaswell commented:
> I would prefer to push this as is, and modify the fipsinstall stuff with a follow on PR since this PR is now otherwise ready-to-merge.

#11580
